File: milkyway-main/lib/home/model/product_model.dart
import 'dart:convert';

class ProductModel {
  int? id;
  String? name;
  String? weight;
  String? price;
  int? isFavourite;
  int? isDaily;
  String? description;
  String? rating;
  String? category;
  List<String> relatedImages = [];
  String? image;
  String? quantity;

  ProductModel(
      {this.name,
      this.weight,
      this.category,
      this.description,
      this.isDaily,
      this.isFavourite,
      this.price,
      this.rating,
      this.image,
      this.id,
      this.quantity,
      required this.relatedImages});

  factory ProductModel.fromJson(Map<String, dynamic> json) {
    return ProductModel(
        id: json["id"],
        relatedImages: (json["relatedImages"] is String)
            ? List<String>.from(jsonDecode(json["relatedImages"]))
            : (json["relatedImages"] is List
                ? List<String>.from(json["relatedImages"])
                : []),
        category: json["category"] ?? "",
        description: json["description"] ?? "",
        isDaily: json["isDaily"],
        isFavourite: json["isFavourite"],
        name: json["name"] ?? "",
        price: json["price"] ?? "",
        rating: json["rating"] ?? "",
        image: json["image"] ?? "",
        quantity: json["quantity"].toString(),
        weight: json["weight"] ?? "");
  }

  Map<String, dynamic> toJson() {
    return {
      "id": id,
      "relatedImages": jsonEncode(relatedImages),
      "name": name,
      "weight": weight,
      "price": price,
      "isFavourite": isFavourite,
      "isDaily": isDaily,
      "description": description,
      "rating": rating,
      "category": category,
      "image": image,
      "quantity": quantity
    };
  }
}

File: Milkyway-Backend/src/models/product.model.js
const mongoose = require('mongoose');

const productSchema = new mongoose.Schema(
    {
        userId: { type: String, index: true },
        id: { type: Number, required: true, index: true },
        name: { type: String, required: true },
        weight: { type: String, default: '' },
        price: { type: String, default: '' },
        isFavourite: { type: Number, default: 0 },
        isDaily: { type: Number, default: 0 },
        description: { type: String, default: '' },
        rating: { type: String, default: '' },
        category: { type: String, default: '' },
        relatedImages: { type: [String], default: [] },
        image: { type: String, default: '' },
        quantity: { type: String, default: '0' },
    },
    { timestamps: true }
);

productSchema.index({ userId: 1, id: 1 });

module.exports = mongoose.model('Product', productSchema);
// const mongoose = require('mongoose');

// const ProductSchema = new mongoose.Schema(
//   {
//     userId: { type: String, index: true },
//     id: { type: Number, required: true, index: true },
//     name: { type: String, required: true },
//     weight: { type: String, default: '' },
//     price: { type: String, default: '' },
//     isFavourite: { type: Number, default: 0 },
//     isDaily: { type: Number, default: 0 },
//     description: { type: String, default: '' },
//     rating: { type: String, default: '' },
//     category: { type: String, default: '' },
//     relatedImages: { type: [String], default: [] },
//     image: { type: String, default: '' },
//     quantity: { type: String, default: '0' },
//   },
//   { timestamps: true }
// );

// ProductSchema.index({ userId: 1, id: 1 }, { unique: false });

// module.exports = mongoose.model('Product', ProductSchema);
